sql/compression_defaults.sql

```diff
@@ -0,0 +1,291 @@
+        GROUP BY 1
+    )
+    SELECT
+      count(*) INTO STRICT _cnt
+    FROM
+      index_attr i
+    INNER JOIN
+      pg_attribute a on (a.attnum = i.attnum AND a.attrelid = relation)
+    LEFT JOIN
+      pg_catalog.pg_attrdef ad ON (ad.adrelid = relation AND ad.adnum = a.attnum)
+    WHERE
+      a.attname NOT IN (SELECT column_name FROM _timescaledb_catalog.dimension d WHERE d.hypertable_id = _hypertable_row.id)
+      AND a.attidentity = '' AND (ad.adbin IS NULL OR pg_get_expr(adbin, adrelid) not like 'nextval%');
+
+    IF _cnt > 0 THEN
+        return json_build_object(
+            'columns', json_build_array(),
+            'confidence', 0,
+            'message',  'Several columns are potential segment by candidates. Please use the segment_by option to explicitly specify the segment_by column');
+    ELSE
+        return json_build_object(
+            'columns', json_build_array(),
+            'confidence', 5,
+            'message',  'You do not have any indexes on columns that can be used for segment_by and thus we are not using segment_by for compression. Please make sure you are not missing any indexes');
+    END IF;
+END
+$BODY$ SET search_path TO pg_catalog, pg_temp;
+
+
+CREATE OR REPLACE FUNCTION _timescaledb_functions.get_orderby_defaults(
+    relation regclass, segment_by_cols text[]
+)
+    RETURNS JSONB LANGUAGE PLPGSQL AS
+$BODY$
+DECLARE
+    _table_name NAME;
+    _schema_name NAME;
+    _hypertable_row _timescaledb_catalog.hypertable;
+    _orderby_names NAME[];
+    _dimension_names NAME[];
+    _first_index_attrs NAME[];
+    _orderby_clauses text[];
+    _confidence int;
+BEGIN
+    SELECT n.nspname, c.relname INTO STRICT _schema_name, _table_name
+    FROM pg_class c
+    INNER JOIN pg_namespace n ON (n.oid = c.relnamespace)
+    WHERE c.oid = relation;
+
+    SELECT * INTO STRICT _hypertable_row FROM _timescaledb_catalog.hypertable h WHERE h.table_name = _table_name AND h.schema_name = schema_name;
+
+    --start with the unique index columns minus the segment by columns
+    with index_attr as (
+        SELECT
+        a.attnum, min(a.pos) as pos
+        FROM
+             --is there a better way to pick the right unique index if there are multiple?
+            (select indkey, indnkeyatts from pg_catalog.pg_index where indisunique and indrelid = relation limit 1) i
+        INNER JOIN LATERAL
+            (select * from unnest(i.indkey) with ordinality) a(attnum, pos) ON (TRUE)
+        WHERE a.pos <= i.indnkeyatts
+        GROUP BY 1
+    )
+    SELECT
+      array_agg(a.attname ORDER BY i.pos) INTO _orderby_names
+    FROM
+      index_attr i
+    INNER JOIN
+      pg_attribute a on (a.attnum = i.attnum AND a.attrelid = relation)
+    WHERE
+      NOT(a.attname::text = ANY (segment_by_cols));
+
+    if _orderby_names is null then
+        _orderby_names := array[]::name[];
+        _confidence := 5;
+    else
+        _confidence := 8;
+    end if;
+
+    --add dimensions
```

Comment:
Probably good to elaborate on this comment. Why do you need to add the dimension?